tests/kernel/tickless/tickless_concept/src/main.c

```diff
@@ -108,13 +126,23 @@ void test_tickless_slice(void)
 	/*enable time slice*/
 	k_sched_time_slice_set(SLICE_SIZE, K_PRIO_PREEMPT(0));
 
+	/*synchronize to tick boundary*/
+	k_sleep(1);
```

Comment:
FWIW: you could consider k_sleep(0) instead, which is guaranteed to synchronize to the "next" tick boundary and doesn't try to sleep for a full millisecond first.   Not particularly relevant here, but in situations where you want to do a bunch of tick synchronization in a tight loop on systems where the tick rate is more than 1khz it might save some time running the test.

Reply:
Good hint. Thank you. Fixed.

Reply:
I changed the code, run the test and it failed.
The k_sleep(0) does not synchronize to tick due to:
```
        /* wait of 0 ms is treated as a 'yield' */
        if (duration == 0) {
                k_yield();
                return 0;
        }
```
I kept k_sleep(1) in the code in order to perform synchronization.